ql/src/java/org/apache/hadoop/hive/ql/plan/ImportTableDesc.java

```diff
@@ -381,4 +382,11 @@ public void setOwnerName(String ownerName) {
         throw new RuntimeException("Invalid table type : " + getDescType());
     }
   }
+
+  public Long getReplWriteId() {
+    if (this.createTblDesc != null) {
+      return this.createTblDesc.getReplWriteId();
```

Comment:
This replWriteId is just a place holder for the writeId from the event message. It need not be in CreateTableDesc. Can be maintained in local variables and pass around.

Reply:
In getBaseAddPartitionDescFromPartition() where we use this function, we don't have access to the event message. Instead we are passing the writeId through ImportTableDesc by calling setReplWriteId(). This function just introduces the missing getReplWriteId() method symmetric to setReplWriteId(). If we use local variable and pass it around there is a possibility that local writeId variable can go out of sync with that in ImportTableDesc.

Reply:
I meant, prepareImport already takes writeId as input parameter which is being set in CreateTableDesc and later read back by getBaseAddPartitionDescFromPartition. Instead, writeId itself can be passed to getBaseAddPartitionDescFromPartition.

Reply:
AFAIU, the reason we set replWriteId in CreateTableDesc is it can be then passed everywhere CreateTableDesc is used. It's better not to create two paths for passing same writeId, with a risk of those going of sync with each other.

Reply:
If we unify writeId and replWriteId in CreateTableDesc, then it's fine.